cloudinit/url_helper.py

```diff
@@ -33,6 +43,7 @@
 LOG = logging.getLogger(__name__)
 
 REDACTED = "REDACTED"
+ExceptionCallback = Optional[Callable[["UrlError"], bool]]
```

Comment:
My goal here is to unify the exception_cb interface used in readurl and wait_for_url as they are currently different.

None of the pre-existing callbacks used the "request_args" argument, and it can also be retrieved from an HTTPError if needed in the future, and calculated it would be harder in `wait_for_url()`, so I removed it from the callback interface and updated the call points accordingly.